### PR TITLE
Add vqueue test for Exists and GetVector method

### PR DESCRIPTION
**Pull request by hlts2**
Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

### Description:

Since there may be a lot of vqueue tests, I will split the PR into multiple parts to make it easier to review.
I added vqueue test for Exists and GetVector method in this PR.
The following are test target function.

```
- IVExists(uuid string) bool
- DVExists(uuid string) bool
- GetVector(uuid string) ([]float32, bool)
```

### Environment:

- Go Version: 1.16.6
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.3

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pkg/agent/core/ngt/service/vqueue/queue.go [hunk 1]
```diff
@@ -246,6 +246,9 @@ func (v *vqueue) RangePopDelete(ctx context.Context, f func(uuid string) bool) {
 	v.flushAndRangeDelete(f)
 }
 
+// GetVector returns the vector stored in the queue.
+// If the same uuid exists in the insert and delete queue, the timestamp is compared.
+// And the vector is returned if the timestamp in the insert queue is newer than the delete queue.
```
Comment (hlts2) [inline pkg/agent/core/ngt/service/vqueue/queue.go:251]:
```suggestion
// GetVector returns the vector stored in the queue.
// If the same UUID exists in the insert queue and the delete queue, the timestamp is compared.
// And the vector is returned if the timestamp in the insert queue is newer than the delete queue.
```

File: pkg/agent/core/ngt/service/vqueue/queue.go [hunk 2]
```diff
@@ -264,6 +267,9 @@ func (v *vqueue) GetVector(uuid string) ([]float32, bool) {
 	return nil, false
 }
 
+// IVExists returns true if there is uuid in the insert queue.
+// If the same uuid exists in the insert and delete queue, the timestamp is compared.
+// And the true is returned if the timestamp in the insert queue is newer than the delete queue.
```
Comment (hlts2) [inline pkg/agent/core/ngt/service/vqueue/queue.go:272]:
```suggestion
// IVExists returns true if there is the UUID in the insert queue.
// If the same UUID exists in the insert queue and the delete queue, the timestamp is compared.
// And the true is returned if the timestamp in the insert queue is newer than the delete queue.
```

File: pkg/agent/core/ngt/service/vqueue/queue.go [hunk 3]
```diff
@@ -280,6 +286,9 @@ func (v *vqueue) IVExists(uuid string) bool {
 	return di <= vec.date
 }
 
+// DVExists returns true if there is uuid in the delete queue.
+// If the same uuid exists in the insert and delete queue, the timestamp is compared.
+// And the true is returned if the timestamp in the delete queue is newer than the insert queue.
```
Comment (hlts2) [inline pkg/agent/core/ngt/service/vqueue/queue.go:291]:
```suggestion
// DVExists returns true if there is the UUID in the delete queue.
// If the same UUID exists in the insert queue and the delete queue, the timestamp is compared.
// And the true is returned if the timestamp in the delete queue is newer than the insert queue.
```

File: pkg/agent/core/ngt/service/vqueue/queue_test.go [hunk 4]
```diff
@@ -884,67 +884,309 @@ func Test_vqueue_GetVector(t *testing.T) {
...
+					date: 4000000000,
+				},
+			}
+			var udim udim
+			for _, key := range udk {
+				udim.Store(key.uuid, key.date)
+			}
+
+			return test{
+				name: "return (1, true) when the uiid dose not exit in udim",
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/vqueue/queue_test.go:996]:
```suggestion
				name: "return (1, true) when the uiid dose not exist in udim",
```

Comment (kevindiu) [reply pkg/agent/core/ngt/service/vqueue/queue_test.go:996]: please replace all `exit` to `exist`

File: pkg/agent/core/ngt/service/vqueue/queue_test.go [hunk 5]
```diff
@@ -1029,67 +1271,301 @@ func Test_vqueue_IVExists(t *testing.T) {
...
+					date: 3000000000,
+				},
+			}
+			var uiim uiim
+			for _, idx := range uii {
+				uiim.Store(idx.uuid, idx)
+			}
+
+			return test{
+				name: "return false when the uiid dose not exit in uiim",
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/vqueue/queue_test.go:1292]:
```suggestion
				name: "return false when the uiid dose not exist in uiim",
```

File: pkg/agent/core/ngt/service/vqueue/queue_test.go [hunk 6]
```diff
@@ -1029,67 +1271,301 @@ func Test_vqueue_IVExists(t *testing.T) {
...
+					date: 4000000000,
+				},
+			}
+			var udim udim
+			for _, key := range udk {
+				udim.Store(key.uuid, key.date)
+			}
+
+			return test{
+				name: "return true when the uiid dose not exit in udim",
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/vqueue/queue_test.go:1380]:
```suggestion
				name: "return true when the uiid dose not exist in udim",
```

File: pkg/agent/core/ngt/service/vqueue/queue_test.go [hunk 7]
```diff
@@ -1174,67 +1650,299 @@ func Test_vqueue_DVExists(t *testing.T) {
...
-		           iBufSize: 0,
-		           dBufSize: 0,
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+			return test{
+				name: "return false when the uiid dose not exit in udim",
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/vqueue/queue_test.go:1671]:
```suggestion
				name: "return false when the uiid dose not exist in udim",
```

File: pkg/agent/core/ngt/service/vqueue/queue_test.go [hunk 8]
```diff
@@ -1174,67 +1650,299 @@ func Test_vqueue_DVExists(t *testing.T) {
...
+					date:   4000000000,
+				},
+			}
+			var uiim uiim
+			for _, idx := range uii {
+				uiim.Store(idx.uuid, idx)
+			}
+
+			return test{
+				name: "return true when the uiid dose not exit in uiim",
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/vqueue/queue_test.go:1759]:
```suggestion
				name: "return true when the uiid dose not exist in uiim",
```

**Review by kevindiu — APPROVED** [review]
LGTM

**Pull request closed without merging**